desktop/cmp/input/DateInput.js

```diff
@@ -308,26 +328,35 @@ class Model extends HoistInputModel {
         return date;
     }
 
-    getFormat() {
-        const {timePrecision} = this,
-            {formatString} = this.props;
+    getFormatString() {
+        const {formatString} = this.props;
+
+        return formatString ?? this.addTimeFmt('YYYY-MM-DD');
+    }
+
+    getParseStrings() {
+        const {formatString, parseStrings} = this.props;
+
+        if (parseStrings) return parseStrings;
+
+        const ret = ['YYYY-MM-DD', 'YYYYMMDD', 'YYYY/MM/DD', 'YYYY-M-DD', 'YYYY/M/DD'].map(s => this.addTimeFmt(s));
+        if (formatString && !ret.includes(formatString)) ret.unshift(formatString);
 
-        if (formatString) return formatString;
-        let ret = 'YYYY-MM-DD';
-        if (timePrecision === 'minute') {
-            ret += ' HH:mm';
-        } else if (timePrecision === 'second') {
-            ret += ' HH:mm:ss';
-        }
         return ret;
     }
 
+    addTimeFmt(fmt) {
+        if (this.timePrecision === 'minute') return fmt + ' HH:mm';
+        if (this.timePrecision === 'second') return fmt + ' HH:mm:ss';
+        return fmt;
```

Comment:
This keeps us in 24-hour time for now, which is the status quo. Could imagine also supporting `hh:mm:ssa` variants.